### PR TITLE
Adding link to all the metrics

### LINKED ISSUE #2437: Add an explanation on how metrics work with engine
## 📚 Documentation

It is unclear for newcomers what does the code below from docs (for example, Accuracy)
```python
metric = Accuracy()
metric.attach(default_evaluator, "accuracy")
y_true = torch.Tensor([1, 0, 1, 1, 0, 1])
y_pred = torch.Tensor([1, 0, 1, 0, 1, 1])
state = default_evaluator.run([[y_pred, y_true]])
print(state.metrics["accuracy"])
```
More precisely, it is unclear that the metric is computed online and at each iteration some internal variables are updated and on epoch completed metric value is computed.

For each metric docs, let's add a link on the explanation that we can add explicitly to https://pytorch.org/ignite/metrics.html

### PR DESCRIPTION
**Pull request by sayantan1410**
Fixes #2437 
Hey, I have made the changes only for the Accuracy, once it is done correctly I will update it for all the metrics in that style. 
Check list:
- [X] New doc strings: description and/or example code are in RST format

### CODE REVIEW

File: ignite/metrics/accuracy.py [hunk 1]
```diff
@@ -116,6 +116,9 @@ class Accuracy(_BaseClassification):
             default, CPU.
 
     Examples:
+        Here, we first declare the metric which is accuracy in this case, then we attach the metric instance to the
```
Comment (vfdev-5) [inline ignite/metrics/accuracy.py:119]: Maybe we can just say simply, for more info how metric works with engine see the link

**Review by vfdev-5 — COMMENTED** [review]
Looks great @sayantan1410 ! 
Let's now add that all metrics. Can you try to put it into defaults.rst before toggle part ?
cc @DevPranjal

**Comment by DevPranjal** [comment]
Yes. That would reduce redundancy in the docstrings.

The [defaults.rst](https://github.com/pytorch/ignite/blob/master/docs/source/defaults.rst) file is included at the start of examples for almost all the metrics (except those in [/ignite/metrics/nlp](https://github.com/pytorch/ignite/tree/master/ignite/metrics/nlp))

**Comment by sayantan1410** [comment]
@sayantan1410 Yeah, I will try to do that

**Comment by sayantan1410** [comment]
Hey, I have done the work related to it in a different PR that is why I am closing this. #2468, this is the extension PR. Extremely sorry for the inconvenience.

**Pull request closed without merging**